core/src/main/java/org/apache/spark/util/collection/unsafe/sort/PrefixComparators.java

```diff
@@ -69,6 +69,8 @@ public static long computePrefix(byte[] bytes) {
      * details see http://stereopsis.com/radix.html.
      */
     public static long computePrefix(double value) {
+      // normalize -0.0 to 0.0, as they should be equal
+      value = value == -0.0 ? 0.0 : value;
```

Comment:
Technically, `value == -0.0` is `true` for both `0.0` and `-0.0`, but the current one is the simplest one for this normalization. We don't need to exclude `0.0` here. +1 for the logic.